### ISSUE TITLE
Crash when running 'terraform state show'

### ISSUE THREAD

**Issue by robothor** [hashicorp/terraform#8315]
Hi folks,

Sorry for another crash report.  This time I was running `terraform state show` on a module.  I hit <return> before I finished typing the entire resource.

### Terraform Version
`terraform -v` shows "Terraform v0.7.0", but the crash log indicates 0.7.0 rc4.  Not sure what's up with that.

### Debug Output
https://gist.github.com/robothor/48f82eeb9539d27d4fe5de3ce2f84535

### Panic Output
https://gist.github.com/robothor/99e0a77abb694e70b5d41f98513c71e9

### Expected Behavior
I expected an error message if I entered a bad argument, not a crash.

### Actual Behavior
The crash above

### Steps to Reproduce
I ran `terraform state show` with the path to the root of a module and not to the actual resource.

e.g. `~/bin/terraform state show module.consul_server_cluster` rather than `~/bin/terraform state show module.consul_server_cluster.aws_route53_record.consul_direct`

### References

Maybe https://github.com/hashicorp/terraform/issues/8255

Please let me know if I can provide more information.

Matt

**Comment by jen20** [comment]
Hi @robothor! This is definitely a bug, however it's somewhat unclear to me whether it's in 0.7 or in 0.7RC4. The steps to reproduce should allow us to determine whether this is still present or not though - thanks for reporting.

**Comment by robothor** [comment]
@jen20: Thanks for looking into this.  It appears that it is probably fixed in 0.7.1.  I had a much more reasonable error after updating today:

~~~
$ terraform state show module.consul_server_cluster
Multiple instances found for the given pattern!

This command requires that the pattern match exactly one instance
of a resource. To view the matched instances, use "terraform state list".
Please modify the pattern to match only a single instance.
exit status 1
~~~

**Comment by apparentlymart** [comment]
Thanks for the follow up, @robothor! I'm going to close this out now.
